feature/experimental/gribi/ate_tests/route_addition_during_failover_test/route_addition_during_failover_test.go

```diff
@@ -0,0 +1,671 @@
+	return secondaryBeforeSwitch
+}
+
+// TestRouteAdditionDuringFailover is to test gRIBI route addition and slave switchover
+// concurrently, validate reinject of gRIBI programmed routes and traffic.
+func TestRouteAdditionDuringFailover(t *testing.T) {
+	d := &oc.Root{}
+	dut := ondatra.DUT(t, "dut")
+	ate := ondatra.ATE(t, "ate")
+
+	ctx := context.Background()
+	gribic := dut.RawAPIs().GRIBI().Default(t)
+	dp1 := dut.Port(t, "port1")
+	ap1 := ate.Port(t, "port1")
+	top := ate.Topology().New()
+	// configure DUT port#1 - source port.
+	configureSubinterfaceDUT(t, d, dp1, 0, 0, dutPort1.IPv4)
+	configureInterfaceDUT(t, dp1, d, "src")
+	configureATE(t, top, ap1, "src", 0, dutPort1.IPv4, atePort1.IPv4CIDR())
+	pushConfig(t, dut, dp1, d)
+	dp2 := dut.Port(t, "port2")
+	ap2 := ate.Port(t, "port2")
+	// Configure 64 subinterfaces on DUT-ATE- PORT#2.
+	subIntfIPs := generateSubIntfPair(t, dut, dp2, ate, ap2, top, d)
+	top.Push(t).StartProtocols(t)
+
+	sysConfigTime := gnmi.Get(t, dut, gnmi.OC().Interface(dut.Port(t, "port1").Name()).LastChange().State())
+
+	// Connect gRIBI client to DUT referred to as gRIBI - using PRESERVE persistence and
+	// SINGLE_PRIMARY mode, with FIB ACK requested. Specify gRIBI as the leader.
+	client := fluent.NewClient()
+	client.Connection().WithStub(gribic).WithPersistence().WithInitialElectionID(1, 0).
+		WithFIBACK().WithRedundancyMode(fluent.ElectedPrimaryClient)
+	client.Start(ctx, t)
+	defer client.Stop(t)
+
+	defer func() {
+		// Flush all entries after test.
+		if err := gribi.FlushAll(client); err != nil {
+			t.Error(err)
+		}
+	}()
+
+	client.StartSending(ctx, t)
+	if err := awaitTimeout(ctx, client, t, time.Minute); err != nil {
+		t.Fatalf("Await got error during session negotiation for clientA: %v", err)
+	}
+	eID := gribi.BecomeLeader(t, client)
+
+	t.Log("Flush existing gRIBI routes before test.")
+	if err := gribi.FlushAll(client); err != nil {
+		t.Fatal(err)
+	}
+
+	args := &testArgs{
+		ctx:        ctx,
+		client:     client,
+		dut:        dut,
+		ate:        ate,
+		top:        top,
+		electionID: eID,
+	}
+
+	virtualIPs := createIPv4Entries(t, ipBlock1FlowArgs.ipBlock)
+
+	t.Log("Inject routes from ipBlock1 in default VRF with NHGID: #1.")
+	pushDefaultEntries(t, args, subIntfIPs, virtualIPs, configNhg, !switchover)
+
+	gr, err := args.client.Get().
+		WithNetworkInstance(*deviations.DefaultNetworkInstance).
+		WithAFT(fluent.IPv4).
+		Send()
+	if err != nil {
+		t.Fatalf("Error encountered during gRIBI Get operation: got: %v", err)
+	}
+
+	for _, r := range gr.GetEntry() {
+		entry := r.GetIpv4().Prefix
+		if got := r.GetFibStatus().String(); got != "PROGRAMMED" {
+			t.Fatalf("gRIBI entry %v is not programmed in FIB, got:%v, want: PROGRAMMED", entry, got)
+		}
+	}
+
+	// Send traffic from ATE port-1 to prefixes in ipBlock1 and ensure traffic
+	// flows 100% and reaches ATE port-2.
+	testTraffic(t, *args, createTrafficFlow(t, args.ate, args.top, ipBlock1FlowArgs))
```

Comment:
Can we keep this traffic running for the whole duration (during the switchover and the following steps) of the test? E.g. like it is done in TE-8.2, use separate startTraffic() and stopTraffic() funcs and call the stop after performing the switchover.

Reply:
testcase 8.2 is already validating it, as per 13.1 it is not required.

Reply:
@LimeHat 
Please follow the readme and dont suggest to add things beyond the readme.